importer/handlers/csv/handler.py

```diff
@@ -0,0 +1,184 @@
+        ]
+        if layer.GetGeometryColumn() or self.default_geometry_column_name and ogr.GeometryTypeToName(layer.GetGeomType()) not in ['Geometry Collection', 'Unknown (any)']:
+            # the geometry colum is not returned rom the layer.schema, so we need to extract it manually
+            # checking if the geometry has been wrogly read as string
+            possible_geometry_column_name = ['geom', 'geometry', 'wkt_geom', 'the_geom']
+            possible_latlong_column = ['latitude', 'longitude', 'lat', 'long', 'x', 'y']
+            schema_keys = [x['name'] for x in layer_schema]
+            geom_is_in_schema = (x in schema_keys for x in possible_geometry_column_name)
+            if any(geom_is_in_schema) and layer.GetGeomType() == 100:  # 100 means None so Geometry not found
+                field_name = [x for x in possible_geometry_column_name if x in schema_keys][0]
+                index = layer.GetFeature(1).keys().index(field_name)
+                geom = [x for x in layer.GetFeature(1)][index]
+                class_name = GEOM_TYPE_MAPPING.get(
+                    self.promote_to_multi(
+                        geom.split("(")[0].replace(" ", "").title()
+                    )
+                )
+                layer_schema = [x for x in layer_schema if field_name not in x['name']]
+            elif any(x in possible_latlong_column for x in schema_keys):
+                class_name = GEOM_TYPE_MAPPING.get(self.promote_to_multi('Point'))
+            else:
+                class_name = GEOM_TYPE_MAPPING.get(self.promote_to_multi(ogr.GeometryTypeToName(layer.GetGeomType())))
+
+            layer_schema += [
+                {
+                    "name": layer.GetGeometryColumn() or self.default_geometry_column_name,
+                    "class_name": class_name,
+                    "dim": 2 if not ogr.GeometryTypeToName(layer.GetGeomType()).lower().startswith('3d') else 3
+                }
+            ]
+
+        # ones we have the schema, here we create a list of chunked value
+        # so the async task will handle max of 30 field per task
+        list_chunked = [
+            layer_schema[i: i + 30] for i in range(0, len(layer_schema), 30)
+        ]
+
+        # definition of the celery group needed to run the async workflow.
+        # in this way each task of the group will handle only 30 field
+        celery_group = group(
+            create_dynamic_structure.s(
+                execution_id, schema, dynamic_model_schema.id, overwrite, layer_name
+            )
+            for schema in list_chunked
+        )
+
+        return dynamic_model_schema, celery_group
+
+    def extract_resource_to_publish(self, files, action, layer_name, alternate, **kwargs):
+        if action == exa.COPY.value:
+            return [
+                {
+                    "name": alternate,
+                    "crs": ResourceBase.objects.filter(Q(alternate__icontains=layer_name) | Q(title__icontains=layer_name))
```

Comment:
This filter is dangerous, it can fetch any layer with a similar name or title. I would rather use a LIKE layer_name || '%' and avoid using the title at all.

Reply:
I add the title since is available for all the resources while the name is available only for the layers.
How do you suggest writing the query?

Reply:
@mattiagiupponi 

`ResourceBase.objects.filter(alternate__istartswith=layer_name)`